test/performance/upload_data.py

```diff
@@ -0,0 +1,136 @@
+    response = client.search(
+        body = query,
+        index = index_name
+    )
+    print('\nSearch results:')
+    print(response)
+
+def setup_os_client():
+    host = os.environ['OS_HOST'] # cluster endpoint, for example: my-test-domain.us-east-1.es.amazonaws.com
+    region = os.environ['OS_REGION']
+    credentials = boto3.Session().get_credentials()
+    auth = AWSV4SignerAuth(credentials, region)
+
+    client = OpenSearch(
+        hosts = [{'host': host, 'port': 443}],
+        http_auth = auth,
+        use_ssl = True,
+        verify_certs = True,
+        connection_class = RequestsHttpConnection
+    )
+    return client
+
+def read_metric_data(metric_file_name):
+    with open(metric_file_name) as f:
+        content = yaml.load(f, Loader=yaml.FullLoader)
+        kpi_data = content['kpi']
+    for data in kpi_data:
+        if data['name'] == 'usage':
+            for metric in data['metrics']:
+                if metric['name'] == 'Memory_MiB':
+                    memory_average = metric['average']
+                    memory_maximum = metric['maximum']
+                elif metric['name'] == 'CPU_millicores':
+                    cpu_average = metric['average']
+                    cpu_maximum = metric['maximum']
+    dt_string = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    return {'upload_date': dt_string, 'memory_average' : float(memory_average), 'memory_maximum': float(memory_maximum), 'cpu_average': float(cpu_average), 'cpu_maximum': float(cpu_maximum), 'memory_average_threshold': get_average_threshold_mem(), 'memory_maximum_threshold': get_maximum_threshold_mem(), 'cpu_average_threshold': get_average_threshold_cpu(), 'cpu_maximum_threshold': get_maximum_threshold_cpu()}    
+
+def generate_id():
+    n = 6
+    return ''.join(random.choices(string.ascii_lowercase + string.digits, k=n))
+
+def get_average_threshold_mem():
+    #return 150
+    return os.environ['TEST_PERFORMANCE_AVG_MEMORY']
+
+def get_maximum_threshold_mem():
+    #return 200
+    return os.environ['TEST_PERFORMANCE_MAX_MEMORY']
+
+def get_average_threshold_cpu():
+    #return 20
+    return os.environ['TEST_PERFORMANCE_AVG_CPU']
+
+def get_maximum_threshold_cpu():
+    #return 100
+    return os.environ['TEST_PERFORMANCE_MAX_CPU']
+
+if __name__ == '__main__':
+    index_name = 'sbo-perf-data'
+    client = setup_os_client()
+    #create_index(client, index_name)
+    #delete_index(client, index_name)
+    # delete_a_document(client, index_name, id)
+
+    metric_file_name = os.environ['KPI_YAML_FILE']
+    doc = read_metric_data(metric_file_name)
+    #doc = {'upload_date': '2022-12-14 06:30:30', 'memory_average' : 68.2, 'memory_maximum': 98.2, 'cpu_average': 10.5, 'cpu_maximum': 90.2, 'memory_average_threshold': 150, 'memory_maximum_threshold': 200, 'cpu_average_threshold': 20, 'cpu_maximum_threshold': 100}
+    id = generate_id()
+    print(f"Random Generated ID: {id}")
+    add_document_to_index(client, index_name, id, doc)
+
+
+    
+
+
```

Comment:
This should not be allowed by the Python linters - just realized that `test/performance` is not included - I fixed that in https://github.com/redhat-developer/service-binding-operator/pull/1332, please rebase once it is merged

Reply:
... and please fix the issues found by the linter

Reply:
Addressed issues pointed by pylint in the latest commit. Rebase is pending since https://github.com/redhat-developer/service-binding-operator/pull/1332 is not merged yet.